core/src/main/scala/cats/data/OneAnd.scala

```diff
@@ -123,8 +123,21 @@ final case class OneAnd[F[_], A](head: A, tail: F[A]) {
     s"OneAnd(${A.show(head)}, ${FA.show(tail)})"
 }
 
+private[data] trait OneAndBinCompat0[F[_], A] {
+  val head: A
+  val tail: F[A]
+
+  @deprecated("Kept for binary compatibility", "2.14.0")
```

Comment:
[question] Is `since = "2.14.0"` specification appropriate (is that the next cats version)? Or do I just not need `@deprecated` at all since it's `private[data]`?

Reply:
personally I wouldn't add the deprecated and just add a comment.

Reply:
If a method is superceded by some other method with the same name, we usually not only make it package local, but also remove all `implicit` keywords from its definition, e.g.:
```
private[data] def combine(other: OneAnd[F, A])(F: Alternative[F]): OneAnd[F, A] = ...
```
which makes it less likely to interfere with the new method while preserving binary compatibility.

I personally don't mind marking it `@deprecated` but don't have strong opinion on that – it is not supposed to be called anyway.